File: src/com/eagle/programmar/Template/Template_Expression.java
// Copyright Eagle Legacy Modernization, LLC, 2010-date
// Original author: Steven A. O'Hara, Nov 12, 2015

package com.eagle.programmar.Template;

import com.eagle.programmar.Template.Expressions.Template_AdditiveExpression;
import com.eagle.programmar.Template.Expressions.Template_AndExpression;
import com.eagle.programmar.Template.Expressions.Template_MultiplicativeExpression;
import com.eagle.programmar.Template.Expressions.Template_NegativeExpression;
import com.eagle.programmar.Template.Expressions.Template_NotExpression;
import com.eagle.programmar.Template.Expressions.Template_OrExpression;
import com.eagle.programmar.Template.Expressions.Template_ParenExpression;
import com.eagle.programmar.Template.Expressions.Template_RelationalExpression;
import com.eagle.programmar.Template.Symbols.Template_Identifier_Reference;
import com.eagle.programmar.Template.Terminals.Template_Literal;
import com.eagle.programmar.Template.Terminals.Template_Number;
import com.eagle.tokens.PrecedenceChooser;
import com.eagle.tokens.PrecedenceOperator;
import com.eagle.tokens.PrecedenceOperator.AllowedPrecedence;
import com.eagle.tokens.interfaces.AbstractExpression;

public class Template_Expression extends PrecedenceChooser implements AbstractExpression
{
	protected static OperatorList _operators = new OperatorList();

	public Template_Expression()
	{
		super(_operators);
	}

	public Template_Expression(PrecedenceOperator token, AllowedPrecedence allowed)
	{
		super(_operators, allowed, token.getClass());
	}

	//
	// Note: All fields should stay in @P(#) order. The # determines operator
	// precedence.
	//

	///////////////////////////////////////////////
	// Terminals

	public @P(10) Template_Number number;
	public @P(20) Template_Literal literal;
	public @P(30) Template_Identifier_Reference id;

	///////////////////////////////////////////////////////////////////////////
	// Primary Expressions

	public @P(100) Template_NegativeExpression negativeExpression;
	public @P(110) Template_NotExpression notExpression;
	public @P(120) Template_ParenExpression parenExpression;

	///////////////////////////////////////////////////////////////////////////
	// Binary Expressions

	public @P(1000) Template_MultiplicativeExpression multiplicativeExpression;
	public @P(1010) Template_AdditiveExpression additiveExpression;
	public @P(1020) Template_RelationalExpression relationalExpression;
	public @P(1030) Template_AndExpression andExpression;
	public @P(1040) Template_OrExpression orExpression;
}

File: csrc/com/eagle/programmar/Template/Template_Expression.cs
// ====================================================================================================
// Produced by the Free Edition of Java to C# Converter.
// Purchase a Premium Edition license at:
// https://www.tangiblesoftwaresolutions.com/order/order-java-to-csharp.html
// ====================================================================================================

// Copyright Eagle Legacy Modernization, LLC, 2010-date
// Original author: Steven A. O'Hara, Nov 12, 2015

namespace com.eagle.programmar.Template
{
	using Template_AdditiveExpression = com.eagle.programmar.Template.Expressions.Template_AdditiveExpression;
	using Template_LogicalAndExpression = com.eagle.programmar.Template.Expressions.Template_LogicalAndExpression;
	using Template_MultiplicativeExpression = com.eagle.programmar.Template.Expressions.Template_MultiplicativeExpression;
	using Template_NegativeExpression = com.eagle.programmar.Template.Expressions.Template_NegativeExpression;
	using Template_LogicalNotExpression = com.eagle.programmar.Template.Expressions.Template_LogicalNotExpression;
	using Template_LogicalOrExpression = com.eagle.programmar.Template.Expressions.Template_LogicalOrExpression;
	using Template_ParenExpression = com.eagle.programmar.Template.Expressions.Template_ParenExpression;
	using Template_RelationalExpression = com.eagle.programmar.Template.Expressions.Template_RelationalExpression;
	using Template_Identifier_Reference = com.eagle.programmar.Template.Symbols.Template_Identifier_Reference;
	using Template_Literal = com.eagle.programmar.Template.Terminals.Template_Literal;
	using Template_Number = com.eagle.programmar.Template.Terminals.Template_Number;
	using PrecedenceChooser = com.eagle.tokens.PrecedenceChooser;
	using PrecedenceOperator = com.eagle.tokens.PrecedenceOperator;
	using AllowedPrecedence = com.eagle.tokens.PrecedenceOperator.AllowedPrecedence;
	using AbstractExpression = com.eagle.tokens.interfaces.AbstractExpression;

	public class Template_Expression : PrecedenceChooser, AbstractExpression
	{
		protected internal static OperatorList _operators = new OperatorList();

		public Template_Expression() : base(_operators)
		{
		}

		public Template_Expression(PrecedenceOperator token, PrecedenceOperator.AllowedPrecedence allowed) : base(_operators, allowed, token.GetType())
		{
		}

		//
		// Note: All fields should stay in @P(#) order. The # determines operator
		// precedence.
		//

		///////////////////////////////////////////////
		// Terminals

// JAVA TO C# CONVERTER TASK: Most Java annotations will not have direct .NET equivalent attributes:
// ORIGINAL LINE: public @P(10) com.eagle.programmar.Template.Terminals.Template_Number number;
		public Template_Number number;
// JAVA TO C# CONVERTER TASK: Most Java annotations will not have direct .NET equivalent attributes:
// ORIGINAL LINE: public @P(20) com.eagle.programmar.Template.Terminals.Template_Literal literal;
		public Template_Literal literal;
// JAVA TO C# CONVERTER TASK: Most Java annotations will not have direct .NET equivalent attributes:
// ORIGINAL LINE: public @P(30) com.eagle.programmar.Template.Symbols.Template_Identifier_Reference id;
		public Template_Identifier_Reference id;

		///////////////////////////////////////////////////////////////////////////
		// Primary Expressions

// JAVA TO C# CONVERTER TASK: Most Java annotations will not have direct .NET equivalent attributes:
// ORIGINAL LINE: public @P(100) com.eagle.programmar.Template.Expressions.Template_NegativeExpression negativeExpression;
		public Template_NegativeExpression negativeExpression;
// JAVA TO C# CONVERTER TASK: Most Java annotations will not have direct .NET equivalent attributes:
// ORIGINAL LINE: public @P(110) com.eagle.programmar.Template.Expressions.Template_LogicalNotExpression notExpression;
		public Template_LogicalNotExpression notExpression;
// JAVA TO C# CONVERTER TASK: Most Java annotations will not have direct .NET equivalent attributes:
// ORIGINAL LINE: public @P(120) com.eagle.programmar.Template.Expressions.Template_ParenExpression parenExpression;
		public Template_ParenExpression parenExpression;

		///////////////////////////////////////////////////////////////////////////
		// Binary Expressions

// JAVA TO C# CONVERTER TASK: Most Java annotations will not have direct .NET equivalent attributes:
// ORIGINAL LINE: public @P(1000) com.eagle.programmar.Template.Expressions.Template_MultiplicativeExpression multiplicativeExpression;
		public Template_MultiplicativeExpression multiplicativeExpression;
// JAVA TO C# CONVERTER TASK: Most Java annotations will not have direct .NET equivalent attributes:
// ORIGINAL LINE: public @P(1010) com.eagle.programmar.Template.Expressions.Template_AdditiveExpression additiveExpression;
		public Template_AdditiveExpression additiveExpression;
// JAVA TO C# CONVERTER TASK: Most Java annotations will not have direct .NET equivalent attributes:
// ORIGINAL LINE: public @P(1020) com.eagle.programmar.Template.Expressions.Template_RelationalExpression relationalExpression;
		public Template_RelationalExpression relationalExpression;
// JAVA TO C# CONVERTER TASK: Most Java annotations will not have direct .NET equivalent attributes:
// ORIGINAL LINE: public @P(1030) com.eagle.programmar.Template.Expressions.Template_LogicalAndExpression andExpression;
		public Template_LogicalAndExpression andExpression;
// JAVA TO C# CONVERTER TASK: Most Java annotations will not have direct .NET equivalent attributes:
// ORIGINAL LINE: public @P(1040) com.eagle.programmar.Template.Expressions.Template_LogicalOrExpression orExpression;
		public Template_LogicalOrExpression orExpression;
	}

}
